ManPhamQuang/COSC2769-rmit
# Feat/Reset password by email

### Change-log
- Introduce pages `Forgot password` & `Reset Password` to allow users to reset their password
- Consume APIs `users/request-reset-password` & `users/reset-password`
- Display success/error message according to server's response
- Cleanup useEffect hook for `NavBar` component

### How-to-test
 - Checkout this branch and start frontend, backend server
 - Log out your current account
 - Go to Signup/ Login page then click the `Forgot password` link to navigate to Forgot Password page
 - Use your registered email to check if the users can receive the reset password link in their mailbox (You can use a temporary email from https://temp-mail.org/en/view/6132eb3d46c9f5011919f849 if don't want to use your private email for receiving the reset link)
 - Change the Reset Password URL from `https://cosc-2769-rmit.vercel.app/user/reset-password?` to `localhost:3000/user/reset-password?`
 - Test multiple cases to ensure that 
- [ ] User can not request reset password with an unregistered email
- [ ] User can not use an expired link to reset password (exp. time 15 mins)
- [ ] Users are not able to access page `/user/forgot-password` OR `/user/reset-password` after logging in

<img width="1434" alt="Screen Shot 2021-09-04 at 10 42 57" src="https://user-images.githubusercontent.com/23531403/132081849-83403a61-3a9d-4e15-9359-fe0b27a22104.png">

<img width="1421" alt="Screen Shot 2021-09-04 at 10 44 02" src="https://user-images.githubusercontent.com/23531403/132081862-6a122821-2b8e-47b2-899e-013499c9cb83.png">
